lib/src/facts/solaris/dmi_resolver.cc

```diff
@@ -67,16 +67,10 @@ namespace facter { namespace facts { namespace solaris {
                 return result.chassis_type.empty() || result.chassis_asset_tag.empty();
             });
         } else if (isa && isa->value() == "sparc") {
-            static boost::regex manufacturer_re("^System Configuration: (.+) sun\\d.$");
-            static boost::regex product_name_re("^SUNW,(.*)$");
+            static boost::regex product_name_manufacturer_re("^System Configuration:\\s+(.+?)\\s+sun\\d+\\S+\\s+(.+)");
             // prtdiag is not implemented in all sparc machines, so we cant get product name this way.
```

Comment:
My $.02 is to just remove the comment as it's no longer applicable (and thus might confuse the reader).